### ISSUE TITLE
Inconsistency between sharing_share_folder docs and implementation

### ISSUE THREAD

**Issue by ianthetechie** [dropbox/dropbox-sdk-python#74]
I found a few issues when attempting to programmatically share folders. I'm not sure whether the issue lies in the implementation, the [documentation](http://dropbox-sdk-python.readthedocs.io/en/master/moduledoc.html#dropbox.dropbox.Dropbox.sharing_share_folder), or a bit of both, so I'll enumerate my findings here and let you tag it appropriately.

The documentation for `sharing_share_folder()` indicates that if an async job ID is returned (by the way, the link here appears to be broken; it took me a while to figure out what's going on here, and that the return class is still the same), that you should call a method called `check_share_job_status()` in order to get status updates. However, such a method does not exist. I believe the correct method is `sharing_check_share_job_status()`. I believe this is just a simple error in the documentation.

Additionally, the `sharing_check_share_job_status()` documentation indicates that the return value will be of type `dropbox.sharing.ShareFolderJobStatus`. However, in the latest library distribution on pip (6.7.0), the returned object is of type `dropbox.sharing.JobStatus`. This seems a bit problematic, as this class has no `get_complete()` method or similar mechanism to get the result (it looks like a very high level abstraction; probably a base class for the other status classes).

I also attempted to inspect the original `dropbox.sharing.ShareFolderLaunch` value, and after the job is complete, the original object is not updated (calling `is_complete()` yields `False`). I believe is the intended behavior, so I'm just mentioning this for completeness. I'm not sure how else to efficiently get the metadata for the newly shared folder.

**Comment by greg-db** [comment]
Thanks for the report!

That does look like a mistake in the [`sharing_share_folder`](https://dropbox-sdk-python.readthedocs.io/en/master/moduledoc.html#dropbox.dropbox.Dropbox.sharing_share_folder) method. We'll look into it.

Also, I just tried it, and `sharing_check_share_job_status` is returning a `ShareFolderJobStatus` for me. Are you maybe accidentally calling `sharing_check_job_status` instead? (Note the missing word "share".) 

Finally, when using `sharing_check_share_job_status`, the returned `ShareFolderJobStatus` should have either `is_complete()` or `is_async_job_id()` yielding `True`. If it's complete, you can use `get_complete()` to get the metadata. If it's an async job, you can poll `sharing_check_share_job_status` to get the status. Once the returned `ShareFolderJobStatus` is complete, you can use `get_complete` on that to get the metadata. (It looks like some of this is missing in the docs, so we'll look into getting that fixed up too.)

**Comment by ianthetechie** [comment]
*facepalm* yes, I am calling `sharing_check_job_status` instead of `sharing_check_share_job_status`. I guess I'd now put in a feature request for a less confusing API naming convention here :P That really threw me. Thanks for pointing that out.

Yes, I was able to determine the correct pattern (sans calling the correct method name) once I realized where to look.

Thanks again for the quick response!
